.github/workflows/prepare-custom-action.yml

```diff
@@ -0,0 +1,51 @@
+name: prepare-custom-action
+
+on:
+  schedule:
+    - cron: '0 0 * * 5'
+
+  push:
+    branches: [ "main" ]
+    paths:
+      - "Dockerfile"
+      - "entrypoint.sh"
+    
+  workflow_dispatch:
+
+env:
+  REGISTRY: ghcr.io
+  IMAGE_NAME: ${{ github.repository }}
+
+jobs:
+  build:
+    runs-on: ubuntu-22.04
+    permissions:
+      contents: read
+      packages: write
+    defaults:
+      run:
+        shell: bash
+
+    steps:
+      - uses: actions/checkout@v3
+
+      - name: Log in to the Container registry
+        uses: docker/login-action@65b78e6e13532edd9afa3aa52ac7964289d1a9c1
+        with:
+          registry: ${{ env.REGISTRY }}
+          username: ${{ github.actor }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Extract metadata (tags, labels) for Docker
+        id: meta
+        uses: docker/metadata-action@9ec57ed1fcdbf14dcef7dfbe97b2010124a938b7
+        with:
+          images: ${{ env.REGISTRY }}/${{ env.IMAGE_NAME }}
+
+      - name: Build and push Docker image
+        uses: docker/build-push-action@f2a1d5e99d037542a71f64918e516c093c6f3fc4
```

Comment:
Although I have had a similar intention, since we publish to GitHub Packages, I have taken GitHub's documentation's recommended flow (since each container could have its details). I would favor keeping the hash per documentation if there are no particular known issues with these that might impact us. https://docs.github.com/en/actions/publishing-packages/publishing-docker-images